web_m2x_options/static/src/js/form.js

```diff
@@ -259,6 +314,54 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
             });
 
             return def;
+        },
+
+        update_mru_list: function(){
+            var self = this,
+                mru_option = 'web_m2x_options_mru';
+            var key = self.compute_mru_key();
+
+            // check if the localstorage has some items for the current model
+            if (localStorage.getItem(mru_option)) {
+                var restore_mru_list = JSON.parse(localStorage.getItem(mru_option));
+                if (restore_mru_list[key]) {
+                    var queue = restore_mru_list[key];
+                    if (queue.indexOf(self.get_value(true)) < 0 && self.get_value(true)){
+                        if (queue.length < 5) {
+                            queue.push(self.get_value(true));
+                        }else {
+                            queue.shift();
+                            queue.push(self.get_value(true));
+                        }
+                        restore_mru_list[key] = queue;
+                    }
```

Comment:
Nope it will stay in place, you referring that the most recurrent one
should be on top?
On Wed, Sep 14, 2016 at 5:32 PM Stéphane Bidoul (ACSONE) <
notifications@github.com> wrote:

> In web_m2x_options/static/src/js/form.js
> <https://github.com/acsone/web/pull/4#discussion_r78785062>:
>
> > +            var key = self.compute_mru_key();
> > +
> > +            // check if the localstorage has some items for the current model
> > +            if (localStorage.getItem(mru_option)) {
> > +                var restore_mru_list = JSON.parse(localStorage.getItem(mru_option));
> > +                if (restore_mru_list[key]) {
> > +                    var queue = restore_mru_list[key];
> > +                    if (queue.indexOf(self.get_value(true)) < 0 && self.get_value(true)){
> > +                        if (queue.length < 5) {
> > +                            queue.push(self.get_value(true));
> > +                        }else {
> > +                            queue.shift();
> > +                            queue.push(self.get_value(true));
> > +                        }
> > +                        restore_mru_list[key] = queue;
> > +                    }
>
> if it's already in the queue, then it should be removed and reinserted on
> top?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/acsone/web/pull/4/files/1ab5431bcf13c489922e34a1ea8c6f97bb38dcfc#r78785062>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANkfciG82hE6aWYUnO1n8ooTvYdF_ehks5qqCG1gaJpZM4J8-Ur>
> .
>